clang/lib/AST/Type.cpp

```diff
@@ -3767,6 +3767,57 @@ void DependentDecltypeType::Profile(llvm::FoldingSetNodeID &ID,
   E->Profile(ID, Context, true);
 }
 
+PackIndexingType::PackIndexingType(const ASTContext &Context,
+                                   QualType Canonical, QualType Pattern,
+                                   Expr *IndexExpr,
+                                   ArrayRef<QualType> Expansions)
+    : Type(PackIndexing, Canonical,
+           computeDependence(Pattern, IndexExpr, Expansions)),
+      Context(Context), Pattern(Pattern), IndexExpr(IndexExpr),
+      Size(Expansions.size()) {
+
+  std::uninitialized_copy(Expansions.begin(), Expansions.end(),
+                          getTrailingObjects<QualType>());
+}
+
+std::optional<unsigned> PackIndexingType::getSelectedIndex() const {
+  if (isInstantiationDependentType())
+    return std::nullopt;
+  // Should only be not a constant for error recovery.
+  ConstantExpr *CE = dyn_cast<ConstantExpr>(getIndexExpr());
+  if (!CE)
```

Comment:
Based on the comment, should this be: `if(getIndexExpr().containsErrors()) return std::nullopt;`, then just a `cast`?

Reply:
The expression itself is fine, but it may not be a constant. Do we think we should set the error flag instead?

Reply:
Why wouldn't the expression be a constant expression here?  The comment claims only when it is an Error Recovery situation (which makes it containsErrors).

Reply:
```
int i = 0;
using foo = pack...[i];
```

We need to somehow parse that (if we fail to parse it it's going to at best not be classified as a type),
but i is not a constant expression. So it's error recovery in the general sense, not a RecoveryExpr

Reply:
Ah, thats what I missed.  I thought you meant that this would be a 'containsError' situation.  I think I'm fine then.

Reply:
If you have idea for a better comment...

Reply:
I don't, but I'll think about it and might comment my next time thru.